File: ch30/CS/App_Code/AppendMessage.cs
using System;
using System.Web;

public class AppendMessage : IHttpModule
{
    private HttpApplication _application = null;

    public void Dispose()
    {
            
    }

    public void Init(HttpApplication context)
    {
        _application = context;
        context.EndRequest += context_EndRequest;
    }

    void context_EndRequest(object sender, EventArgs e)
    {
        string message = string.Format("processed on {0}", System.DateTime.Now.ToString());
        _application.Context.Response.Write(message);
    }
}

File: ch30/VB/App_Code/AppendMessage.vb
Public Class AppendMessage
    Implements IHttpModule

    Dim WithEvents _application As HttpApplication = Nothing

    Public Sub Dispose() Implements IHttpModule.Dispose

    End Sub

    Public Sub Init(context As HttpApplication) Implements IHttpModule.Init
        _application = context
    End Sub

    Public Sub context_EndRequest(ByVal sender As Object, ByVal e As EventArgs) _
            Handles _application.EndRequest

        Dim message As String = String.Format("processed on {0}", System.DateTime.Now.ToString())

        _application.Context.Response.Write(message)
    End Sub

End Class
